web/templates/settings/custom_user_profile_field.hbs

```diff
@@ -2,30 +2,32 @@
         {{else if is_select_field}}
-        <select class="custom_user_field_value settings_select modal_select bootstrap-focus-style">
-            <option value=""></option>
-            {{#each field_choices}}
-                <option value="{{ this.value }}" {{#if this.selected}}selected{{/if}}>{{ this.text }}</option>
-            {{/each}}
-        </select>
+        <div class="select-field-wrapper">
+            <select class="custom_user_field_value settings_select modal_select bootstrap-focus-style {{#unless editable_by_user}}disabled{{/unless}}">
+                <option value=""></option>
+                {{#each field_choices}}
+                    <option value="{{ this.value }}" {{#if this.selected}}selected{{/if}}>{{ this.text }}</option>
+                {{/each}}
+            </select>
+        </div>
         {{else if is_user_field }}
-        <div class="pill-container person_picker">
-            <div class="input" contenteditable="true"></div>
+        <div class="pill-container person_picker {{#unless editable_by_user}}disabled{{/unless}}" >
+            <div class="input" {{#if editable_by_user}}contenteditable="true"{{/if}}></div>
         </div>
         {{else if is_date_field }}
         <input class="custom_user_field_value datepicker modal_text_input" data-field-id="{{ field.id }}" type="text"
-          value="{{ field_value.value }}" />
-        <span class="remove_date"><i class="fa fa-close"></i></span>
+          value="{{ field_value.value }}" {{#unless editable_by_user}}disabled="disabled"{{/unless}}/>
+        {{#if editable_by_user}}<span class="remove_date"><i class="fa fa-close"></i></span>{{/if}}
         {{else if is_url_field }}
-        <input class="custom_user_field_value modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="2048" />
+        <input class="custom_user_field_value modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" {{#unless editable_by_user}}disabled="disabled"{{/unless}} maxlength="2048" />
         {{else if is_pronouns_field}}
-        <input class="custom_user_field_value pronouns_type_field modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="50" />
+        <input class="custom_user_field_value pronouns_type_field modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="50" {{#unless editable_by_user}}disabled="disabled"{{/unless}}/>
         {{else}}
-        <input class="custom_user_field_value modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="50" />
+        <input class="custom_user_field_value modal_text_input" type="{{ field_type }}" value="{{ field_value.value }}" maxlength="50" {{#unless editable_by_user}}disabled="disabled"{{/unless}}/>
         {{/if}}
```

Comment:
Have you checked that this looks visually nice with all of these field types?

Reply:
Yes , I have tested all the scenarios in dark and light mode. It looks consistent with other fields. I have also added screenshots in PR description in case you want to take a quick look.